### ISSUE TITLE
Possible bug: Using test set as validation for M4 (data leakage?)

### ISSUE THREAD

**Issue by Mango123L** [cong-zeng/ood-llm-detect#9]
Hi, thanks for the work.

It seems the current M4 branch uses the test split as validation to select the best model.

Code (train_classifier_dsvdd.py):
```python
elif opt.dataset=='M4':
    dataset = load_M4(opt.path)
    passages_dataset = PassagesDataset(
        dataset[opt.database_name] + dataset[opt.database_name.replace('train','dev')],
        mode='M4'
    )
    val_dataset = PassagesDataset(dataset[opt.test_dataset_name], mode='M4')
```
Default script arguments set:
--database_name multilingual_train
--test_dataset_name multilingual_test

So:
- Training data = multilingual_train + multilingual_dev
- “Validation” (for best model) = multilingual_test (the test set)

**Question:**
Should multilingual_dev be used for validation and multilingual_test only for final evaluation? Thanks!
